### ISSUE TITLE
How do I test that a job was enqueued for later?

### ISSUE THREAD

**Issue by scashin133** [gocardless/rspec-activejob#9]
Specifically the wait and wait_until options of set.

From http://edgeguides.rubyonrails.org/active_job_basics.html#enqueue-the-job
Enqueue a job to be performed tomorrow at noon.
MyJob.set(wait_until: Date.tomorrow.noon).perform_later(record)

Enqueue a job to be performed 1 week from now.
MyJob.set(wait: 1.week).perform_later(record)

**Comment by isaacseymour** [comment]
Good point, hadn't thought about that! I'll add it was a modifier to the `enqueue_a` and `have_been_enqueued` matchers when I have some time :)

**Comment by scashin133** [comment]
Awesome, thanks for the quick response! I'll gladly file a PR tonight after work.

**Comment by macfanatic** [comment]
@scashin133 Did you come up with anything to test this out of curiosity?

Curious, otherwise I'll just do mock expectations I suppose.

**Comment by macfanatic** [comment]
For the time being I wrote my spec as the following:

```ruby
    it 'enqueues a job in the future' do
      expect { subject.run(item) }.to enqueue_a(WorkerJob).with(global_id(item))

      queued_job = ActiveJob::Base.queue_adapter.enqueued_jobs.find do |job_info|
        job_info[:job] == Jobs::WorkerClass
      end

      future = Time.at(queued_job[:at])
      expect(future).to be_within(4.hours).of(Time.now)
    end
```

In my case my time window is a random interval between 1 and 4 hours, so that's essentially all I'm testing here.

**Comment by ngelx** [comment]
This feature will be nice and helpful. Any news about it? 

The solution proposed by @macfanatic is fine, but seems to be overcomplicated when you got more than 3 or 4 jobs to test for a simple condition.

**Comment by isaacseymour** [comment]
Sorry for being so slow on this - I will try to find some time to work on this gem soon (I also need to make it work better when multiple jobs get enqueued by the block). PRs are very welcome if anyone else has the time to implement this!

**Comment by mrfoto** [comment]
Hey, good work with the gem. Are there any updates on this?

**Comment by isaacseymour** [comment]
Sorry @mrfoto - realistically not going to get the time to add this feature any time soon. Always open to PRs though if someone else has the time.

**Comment by mrfoto** [comment]
You should close this issue since it was now done in #19 :tada:

**Comment by isaacseymour** [comment]
Closed by #19 :dancer:
